src/libs/ck-libs/ampi/ampiOneSided.C

```diff
@@ -1021,6 +1021,52 @@ AMPI_API_IMPL(int, MPI_Win_unlock, int rank, MPI_Win win)
   return MPI_SUCCESS;
 }
 
+/*
+ * int AMPI_Win_lock_all(int assert, MPI_Win win)
+ *   Locks access to this MPI_Win object for all ranks.
+ *   Input:
+ *     int assertion : program assertion, used to provide optimization hints
+ *   Returns int : MPI_SUCCESS or MPI_ERR_WIN
+ */
+AMPI_API_IMPL(int, MPI_Win_lock_all, int assert, MPI_Win win)
+{
+  AMPI_API("AMPI_Win_lock_all");
+  WinStruct *winStruct = getAmpiParent()->getWinStruct(win);
+  ampi *ptr = getAmpiInstance(winStruct->comm);
+  int size = ptr->getSize();
+
+  // process assertion here:
```

Comment:
Could maybe change these two lines to "// TODO: optimize for assertions"

Reply:
fixed